csm/core/agent/api.py

```diff
@@ -312,11 +317,26 @@ async def _validate_bearer(login_service: LoginService, session_id: str):
         Log.info(f'Username: {session.credentials.user_id}')
         return session
 
+    @staticmethod
+    def _retrieve_config(request):
+        path = request.path
+        method =  request.method
+        key = method + ":" + path
+        conf_key = CsmAuth.HYBRID_APIS[key]
+        return Conf.get(const.CSM_GLOBAL_INDEX, conf_key)
+
     @staticmethod
     @web.middleware
     async def session_middleware(request, handler):
         session = None
         is_public = await CsmRestApi._is_public(request)
+        is_hybrid = await CsmRestApi._is_hybrid(request)
+        if is_hybrid:
+            conf_key = CsmRestApi._retrieve_config(request)
+            if conf_key == "true" or conf_key == "True":
```

Comment:
Change the logic based on new key mentioned above

Reply:
resolved